agents/plugins/yum

```diff
@@ -1,125 +1,116 @@
+then
+    # something is wrong here...
+    echo "invalid cache file"
+    exit 2
 fi
 
 # check if system has rebooted - if so, remove cached check file to avoid wrong "reboot required"-state
 RUNNING_SECS=$(cat /proc/uptime | cut -d" " -f1 | cut -d"." -f1)
 
 # assume 120 seconds for reboot being enough
-if (( $RUNNING_SECS < 120 ))
+if (( RUNNING_SECS < 120 ))
 then
-    # remove pre-reboot cache which reqires reboot
-    rm $YUM_CACHE_FILE
+    # remove pre-reboot cache which requires reboot
+    rm -f $CACHE_RESULT_CHECK
     # create empty check cache
-    touch $YUM_CACHE_FILE
+    touch $CACHE_RESULT_CHECK
 fi
 
-# get cached information
-YUM_CACHED="$(cat $YUM_CACHE_FILE)"
+echo "<<<yum>>>"
 
-# compare current and cached state
-if [ "$YUM_CURRENT" != "$YUM_CACHED" ] || [ ! -s $YUM_CACHE_FILE ]
+# compare current and cached yum information
+if [ "$YUM_CURRENT" != "$YUM_CACHED)" ] && [ ! -s $CACHE_RESULT_CHECK ]
 then
-    if [ -z $(pgrep -f "python /usr/bin/yum") ]; then
-        LATEST_KERNEL=$(yum -q -C --noplugins --debuglevel 0 list installed | egrep "^(vz)?kernel(|-(uek|ml|lt))\." | grep "\." | tail -n1 | awk '{print $2};')
-        RUNNING_KERNEL=$(cat /proc/version | awk '{print $3}' | sed 's/.x86_64//g')
-        
-        if [[ "$RUNNING_KERNEL" == "$LATEST_KERNEL"* ]]
-        then
-            BOOT_REQUIRED="no"
-        else
-            BOOT_REQUIRED="yes"
-        fi
-
-        UPDATES=$(waitmax 25 /usr/bin/yum -C --noplugins --quiet list updates | grep "\." | cut -d' ' -f1 | wc -l || echo "-1")
-
-	# check if --security is available
-	waitmax 10 /usr/bin/yum -C --noplugins --quiet --security list updates > /dev/null 2>&1
-	if [ $? -eq 0 ]
-	then
-	        SECURITY_UPDATES=$(waitmax 25 /usr/bin/yum -C --noplugins --quiet --security list updates | grep "\." | cut -d' ' -f1 | wc -l || echo "-1")
-	else
-	# --security not supported with this yum version
-	# maybe the yum-plugin-security package is needed (RH 6)
-	        SECURITY_UPDATES="-2"
-	fi
-
-        echo $BOOT_REQUIRED
-        echo $UPDATES
-        echo $SECURITY_UPDATES
-        
-        # check if cached listing of /var/cache/yum already exists
-        if [ ! -L $MK_VARDIR/cache/yum_state.cache ] && [ ! -d $MK_VARDIR/cache/yum_state.cache ]
-        then
-            # cache new yum state for future checks
-            echo "$YUM_CURRENT" > $MK_VARDIR/cache/yum_state.cache
-        else
-            # something is wrong here...
-            exit
-        fi
-                
-        # cache check results
-        # check if cached check result already exists and is nothing but a file
-        if [ -f $YUM_CACHE_FILE ] || [ ! -L $YUM_CACHE_FILE ]
-        then
-            echo $BOOT_REQUIRED > $YUM_CACHE_FILE
-            echo $UPDATES >> $YUM_CACHE_FILE
-            echo $SECURITY_UPDATES >> $YUM_CACHE_FILE
-        else
-            # something is wrong here...
-            echo "invalid check cache file"
-            exit
-        fi
+    count=0
```

Comment:
maybe just use COUNT like all the other variables in CAPITAL letters?

Reply:
Usually I use UPPER for Constant and Environment Variables otherwise it's better to use LOWER.
It's just a matter of style coding

Reply:
Yes, I do it your way in the meantime - this code is somewhat older. Maybe a good momentf or me  to fix this.